Separator Block: add border property for dots style to editor stylesheet

mikachan opened the pull request:
<!-- Thanks for contributing to Gutenberg! Please follow the Gutenberg Contributing Guidelines:
https://github.com/WordPress/gutenberg/blob/trunk/CONTRIBUTING.md -->

## What?
<!-- In a few words, what is the PR actually doing? -->
This PR adds a `border: none` property for the dots style of the separator block to the block's editor.scss file. 

## Why?
<!-- Why is this PR necessary? What problem is it solving? Reference any existing previous issue(s) or PR(s), but please add a short summary here, too -->
Currently, the dots style renders a border below the dots in the editor when border settings have been set in the `theme.json` file.

| Before      | After |
| ----------- | ----------- |
| <img width="657" alt="image" src="https://user-images.githubusercontent.com/1645628/166973592-f721a0f3-f730-4750-a8ba-5e8813a4663e.png">      | <img width="689" alt="image" src="https://user-images.githubusercontent.com/1645628/166973929-a6e27e8b-ecee-43b9-a23a-1e13a2fa1c78.png">       |

## How?
<!-- How is your PR addressing the issue at hand? What are the implementation details? -->
This is happening because the theme.json border CSS for the separator appears at a higher precedence than the `.is-style-dots` CSS. The border property in the editor.scss overrides this.

## Testing Instructions
<!-- Please include step by step instructions on how to test this PR. -->
<!-- 1. Open a Post or Page. -->
<!-- 2. Insert a Heading Block. -->
<!-- 3. etc. -->
1. Activate Twenty Twenty-Two
2. Add the following separator settings to the `theme.json` file:
```
"core/separator": {
    "border": {
        "color": "#000",
        "style": "solid",
        "width": "0 0 1px 0"
    }
}
```
3. Edit a post or page
4. Insert a separator block
5. Change the style of the separator block to the dots style
6. Without this PR, you should see a solid border below the dots in the editor (it displays correctly on the front end)

mikachan committed:
Add !important to border style

joyously commented:
I don't think either property should have `!important`.
Once upon a time, I tried to style for dots in my theme using the background property with radial gradients, but I couldn't because I couldn't override the importance of the base style. 
This is actually the thing that made me give up on trying to support block styles in my theme.

paaljoachim commented:
It would be helpful to not include `!important` as it makes custom styling very difficult.

glendaviesnz commented:
Because this is only an issue in the editor I think it would be better to just fix this in the editor.scss file if we can. 

This could be done without adding an `! important` if a straight `border: none;` is [added here](https://github.com/WordPress/gutenberg/blob/trunk/packages/block-library/src/separator/editor.scss#L9), and this seems to have the required specificity to override the theme.json with a quick test I did.

Although background already has `!important` applied in the `style.scss`, there would be backwards compatibility issues with also adding it to the `border` as there may be themes/plugins that are relying on `border` having its current specificity of (0,2,0).

carolinan commented:
I am not able to reproduce the issue using Twenty-Twenty Two, WordPress 6.0 Rc1 and Gutenberg current trunk (without the PR) Please provide detailed steps on how to test this.

mikachan committed:
Remove !important from style.scss

mikachan committed:
Add border property to editor.scss

mikachan commented:
Thanks all for your feedback, I agree that adding an `!important` isn't ideal.

> Because this is only an issue in the editor I think it would be better to just fix this in the editor.scss file if we can.

I think this is a much better solution, thanks! I've updated this PR in line with this.

> I am not able to reproduce the issue using Twenty-Twenty Two, WordPress 6.0 Rc1 and Gutenberg current trunk (without the PR) Please provide detailed steps on how to test this.

Sorry, I should've added the theme I was testing with. I've updated the testing instructions. I also cannot reproduce this using TT2.

mikachan commented:
> I also cannot reproduce this using TT2.

I've managed to reproduce this using TT2 now, I was missing the crucial theme.json settings that were causing the border property override 🙈

Please try adding the following to TT2's theme.json:

```
"core/separator": {
    "border": {
        "color": "#000",
        "style": "solid",
        "width": "0 0 1px 0"
    },
    "color": {
        "text": "#000"
    }
}
```

glendaviesnz reviewed (approved):
This tested well for me.
